src/test01/services/buttonService.js

```diff
@@ -0,0 +1,22 @@
+export function buttonService() {
+    'use strict';
+
+    angular
+        .module('app')
+        .service('buttonService', ButtonService);
+
+    ButtonService.$inject = ['$http'];
+
+    function ButtonService($http) {
+
+        var _self = this;
+
+        _self.getData = getData;
+
+        function getData() {
+            return $http.get('test01/data.json');
+        }
+
+        return this;
+    }
+};
```

Comment:
This service's sole purpose is to return the JSON object contained in data.json. It does so by using the $http service to request data.json from the web server.